Fix/grpc stream

tang95 opened the pull request:
When `span := trace.SpanFromContext(ctx)` returns a `noopSpan`, `gctx` might be nil, leading to a nil pointer error.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=285) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=285) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/opentelemetry-go-auto-instrumentation?pullRequest=285) it.</sub>

NameHaibinZhang commented:
please signed CLA

123liuziming reviewed (approved):
Could you please add a test case for this, otherwise LGTM, Thanks!

tang95 commented:
Added test cases and moved the changes to a new PR because the commits were messy. Closing this PR. Thanks for your suggestions! https://github.com/alibaba/opentelemetry-go-auto-instrumentation/pull/287
